msvc/scripts/tests/setup-env.bat

```diff
@@ -0,0 +1,47 @@
+@ECHO OFF
+
+SET CURRENT_MONO_HOME=%MONO_HOME%
+IF "" == "%CURRENT_MONO_HOME%" (
+	SET CURRENT_MONO_HOME=%~dp0..\..\..
+)
+
+REM Convert to absolute path.
+pushd %CURRENT_MONO_HOME%
+SET CURRENT_MONO_HOME=%CD%
+popd
+
+
+SET MONO_BUILD_CONFIG=Release
+SET MONO_CFG_DIR=%CURRENT_MONO_HOME%\runtime\etc
+
+SET MONO_BCL_HOME=%CURRENT_MONO_HOME%\mcs\class\lib
+SET MONO_MINI_HOME=%CURRENT_MONO_HOME%\mono\mini
+
+SET MONO_BUILD_DIR=%CURRENT_MONO_HOME%\msvc\build
+SET MONO_AOT_BUILD_DIR=%MONO_BUILD_DIR%\fullaot
+SET MONO_DIST_DIR=%CURRENT_MONO_HOME%\msvc\dist
+
+SET MONO_BCL_PATH=%MONO_BCL_HOME%\net_4_x
+SET MONO_BCL_FACADE_PATH=%MONO_BCL_PATH%\Facades
+SET MONO_WINAOT_BCL_PATH=%MONO_BCL_HOME%\winaot
+SET MONO_WINAOT_BCL_FACADE_PATH=%MONO_WINAOT_BCL_PATH%\Facades
+
+SET MONO_WINAOT_BUILD_DIR=%MONO_AOT_BUILD_DIR%\winaot
+SET MONO_TEST_BUILD_DIR=
+SET MONO_SAMPLE_BUILD_DIR=
+
+SET MONO_CROSS_COMPILER_HOME=%MONO_BUILD_DIR%\sgen\x64\bin\%MONO_BUILD_CONFIG%
+
+REM Look for Mono repositories.
+ECHO Searching for Mono repositories...
+IF NOT EXIST "%CURRENT_MONO_HOME%" (
+	ECHO Could not find "%CURRENT_MONO_HOME%".
+	EXIT /b -1'
```

Comment:
Negative exit codes cause problems.
Some process launchers e.g. Perl interpret them  as command not found instead of a deliberate error, and retry running a slightly altered form -- causing the entire thing to run twice.
Favor small positive exit codes (Posix and Perl etc. also struggle with exit codes over 255).